x/merkledb/proof_test.go

```diff
@@ -1251,31 +1438,51 @@ func TestVerifyProofPath(t *testing.T) {
 				{Key: ToKey([]byte{1, 2, 3})},
 				{Key: ToKey([]byte{1, 2, 3})},
 			},
-			proofKey:    maybe.Some(ToKey([]byte{1, 2, 3})),
+			proofKey:    ToKey([]byte{1, 2, 3}),
 			expectedErr: ErrProofNodeNotForKey,
 		},
 		{
-			name: "oddLength key with value",
+			name: "odd length key with value",
 			path: []ProofNode{
-				{Key: ToKey([]byte{1})},
-				{Key: ToKey([]byte{1, 2})},
 				{
-					Key: Key{
-						value:  string([]byte{1, 2, 240}),
-						length: 20,
-					},
+					Key: ToKey([]byte{1}),
+				},
+				{
+					Key:         ToKey([]byte{1, 2}).Take(12),
+					ValueOrHash: maybe.Some([]byte{1}),
+				},
+				{
+					Key: ToKey([]byte{1, 2, 3}),
+				},
+			},
+			proofKey:    ToKey([]byte{1, 2, 3}),
+			expectedErr: ErrPartialByteLengthWithValue,
+		},
+		{
+			name: "odd length key with value",
+			path: []ProofNode{
+				{
+					Key: ToKey([]byte{1}),
+				},
+				{
+					Key:         ToKey([]byte{1, 2}).Take(12),
 					ValueOrHash: maybe.Some([]byte{1}),
 				},
+				{
+					Key: ToKey([]byte{1, 2, 3}),
+				},
 			},
-			proofKey:    maybe.Some(ToKey([]byte{1, 2, 3})),
+			proofKey:    ToKey([]byte{1, 2, 3}),
 			expectedErr: ErrPartialByteLengthWithValue,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := verifyProofPath(tt.path, tt.proofKey)
-			require.ErrorIs(t, err, tt.expectedErr)
+			require := require.New(t)
+
+			err := verifyProofPath(tt.path, tt.proofKey, tokenSize)
```

Comment:
Since `tokenSize` is only used in this test body can we just define it here? The value is self-documenting since the parameter name is already `tokenSize`.
```suggestion
			err := verifyProofPath(tt.path, tt.proofKey, 8)
```